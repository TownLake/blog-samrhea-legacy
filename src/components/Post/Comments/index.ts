export { default as Comments } from "./Comments";
