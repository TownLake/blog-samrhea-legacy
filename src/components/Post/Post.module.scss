@import "src/assets/scss/variables";
@import "src/assets/scss/mixins";

.post {
  .content {
    @include margin-auto();
  }

  .footer {
    max-width: $layout-post-width;
    @include padding(0, 0.5);
    @include margin-auto();
  }

  .comments {
    max-width: $layout-post-width;
    @include padding(0, 0.5);
    @include margin-auto();
  }

  .buttons {
    align-items: center;
    display: flex;
    justify-content: center;
    text-align: center;
    @include margin(1, 0, 0, 0);

    .buttonArticles {
      @include margin(0, 0.5, 0, 0);
    }
  }
}

@include breakpoint-md {
  .post {
    .footer {
      @include padding-equal(0);
    }

    .comments {
      @include padding-equal(0);
    }

    .buttons {
      left: 30px;
      max-width: none;
      position: fixed;
      top: 30px;
      @include margin(0, 0.5, 0, 0);
    }
  }
}
