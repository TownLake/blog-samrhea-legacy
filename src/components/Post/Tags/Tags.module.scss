@import "../../../assets/scss/variables";
@import "../../../assets/scss/mixins";

.tags {
  @include margin-bottom(0.5);

  .list {
    list-style: none;
    @include padding-equal(0);

    .item {
      display: inline-block;
      @include margin(0.25, 0);

      @include breakpoint-sm {
        &:first-child {
          @include margin-left(0);
          @include padding-left(0);
        }
      }
    }
  }
}
