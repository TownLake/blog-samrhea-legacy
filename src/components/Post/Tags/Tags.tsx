import React from "react";

import { Button } from "@/components/Button";

import * as styles from "./Tags.module.scss";

type Props = {
  tags: string[];
  tagSlugs: string[];
};

const Tags = ({ tags, tagSlugs }: Props) => (
  <div className={styles.tags}>
    <ul className={styles.list}>
      {tagSlugs
        ? tagSlugs.map((slug, i) => (
            <li className={styles.item} key={slug}>
              <Button title={tags[i]} key={slug} to={slug} />
            </li>
          ))
        : null}
    </ul>
  </div>
);

export default Tags;
