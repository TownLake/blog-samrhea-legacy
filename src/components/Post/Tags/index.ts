export { default as Tags } from "./Tags";
