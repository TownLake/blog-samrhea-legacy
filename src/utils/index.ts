export { default as getIcon } from "./get-icon";
export { default as getContactHref } from "./get-contact-href";
export { default as toKebabCase } from "./to-kebab-case";
export { default as getDefaultColorMode } from "./get-default-color-mode";
export * as testUtils from "./test-utils";
