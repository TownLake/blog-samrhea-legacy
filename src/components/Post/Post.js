// @flow strict
import React from 'react';
import { Link } from 'gatsby';
import Author from './Author';
import Comments from './Comments';
import Content from './Content';
import Meta from './Meta';
import Tags from './Tags';
import styles from './Post.module.scss';
import type { Node } from '../../types';

type Props = {
  post: Node
};

const Post = ({ post }: Props) => {
  const { html } = post;
  const { tagSlugs, slug } = post.fields;
  const { tags, title, date } = post.frontmatter;

  return (
    <div className={styles['post']}>
      <Link className={styles['post__home-button']} to="/">Home</Link>

      <div className={styles['post__content']}>
        <Content body={html} title={title} />
      </div>

      <div className={styles['post__footer']}>
        <Meta date={date} />
        {tags && tagSlugs && <Tags tags={tags} tagSlugs={tagSlugs} />}
        <Author />
      </div>

      <div className={styles['post__comments']}>
        <Comments postSlug={slug} postTitle={post.frontmatter.title} />
      </div>
    </div>
  );
};

export default Post;
