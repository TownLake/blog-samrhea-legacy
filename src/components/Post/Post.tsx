import React from "react";

import { Button } from "@/components/Button";
import { ThemeSwitcher } from "@/components/ThemeSwitcher";
import type { Node } from "@/types";

import { Author } from "./Author";
import { Comments } from "./Comments";
import { Content } from "./Content";
import { Meta } from "./Meta";
import { Tags } from "./Tags";

import * as styles from "./Post.module.scss";

interface Props {
  post: Node;
}

const Post: React.FC<Props> = ({ post }: Props) => {
  const { html } = post;
  const { tagSlugs, slug } = post.fields;
  const { tags, title, date } = post.frontmatter;

  return (
    <div className={styles.post}>
      <div className={styles.buttons}>
        <Button className={styles.buttonArticles} title="All Articles" to="/" />
        <ThemeSwitcher />
      </div>

      <div className={styles.content}>
        <Content body={html} title={title} />
      </div>

      <div className={styles.footer}>
        <Meta date={date} />
        {tags && tagSlugs && <Tags tags={tags} tagSlugs={tagSlugs} />}
        <Author />
      </div>

      <div className={styles.comments}>
        <Comments postSlug={slug} postTitle={post.frontmatter.title} />
      </div>
    </div>
  );
};

export default Post;
