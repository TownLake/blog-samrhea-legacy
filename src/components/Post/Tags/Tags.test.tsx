import React from "react";

import { Tags } from "@/components/Post/Tags";
import * as mocks from "@/mocks";
import { testUtils } from "@/utils";

describe("Tags", () => {
  test("renders correctly", () => {
    const props = {
      tags: mocks.markdownRemark.frontmatter.tags,
      tagSlugs: mocks.markdownRemark.fields.tagsSlugs,
    };

    const tree = testUtils
      .createSnapshotsRenderer(<Tags {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
