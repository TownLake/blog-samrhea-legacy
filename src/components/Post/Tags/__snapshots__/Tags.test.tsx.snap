// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags renders correctly 1`] = `
<div>
  <ul>
    <li>
      <a
        className=""
        href="/handwriting"
      >
        Handwriting
      </a>
    </li>
    <li>
      <a
        className=""
        href="/helvetica"
      >
        Helvetica
      </a>
    </li>
  </ul>
</div>
`;
